build/Makefile.deps

```diff
@@ -1,17 +1,60 @@
-#### Dependency targets
 
 ## Installs all the required binaries for the makefile targets.
-.PHONY: deps
-deps:
-ifndef SKIP_DEPCHECK
-	@ echo "-> Installing project dependencies..."
-	@ GO111MODULE=off go get -u github.com/myitcv/gobin
-	@ $(GOBIN)/gobin github.com/go-swagger/go-swagger/cmd/swagger@v0.22.0
-	@ $(GOBIN)/gobin github.com/elastic/go-licenser@v0.3.0
-	@ $(GOBIN)/gobin golang.org/x/lint/golint
-	@ curl -sfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh| sh -s -- -b $(GOBIN) v1.22.2
-	@ echo "-> Done."
-endif
+
+OS:=$(shell uname -s | tr '[:upper:]' '[:lower:]')
+ARCH:=$(shell $(PWD)/scripts/uname_arch.sh)
+
+VERSION_GOBIN:=v0.0.13
+VERSION_GOLINT:=v0.0.0-20191125180803-fdd1cda4f05f
+VERSION_GOLICENSER:=v0.3.0
+VERSION_GOLANGCILINT:=v1.22.0
+VERSION_SWAGGER:=v0.22.0
+
+deps: $(GOBIN)/gobin $(GOBIN)/golint $(GOBIN)/go-licenser $(GOBIN)/golangci-lint $(GOBIN)/swagger
+
+$(GOBIN):
+	@ mkdir -p $(GOBIN)
+
+.version-gobin-$(VERSION_GOBIN):
+	@ rm -f .version-gobin-*
+	@ echo $(VERSION_GOBIN) > .version-gobin-$(VERSION_GOBIN)
+
+$(GOBIN)/gobin: .version-gobin-$(VERSION_GOBIN) | $(GOBIN)
+	@ echo "-> Installing gobin..."
+	@ curl -sL -o $(GOBIN)/gobin https://github.com/myitcv/gobin/releases/download/$(VERSION_GOBIN)/$(OS)-$(ARCH)
+	@ chmod +x $(GOBIN)/gobin
+
+.version-golint-$(VERSION_GOLINT):
+	@ rm -f .version-golint-*
+	@ echo $(VERSION_GOLINT) > .version-golint-$(VERSION_GOLINT)
```

Comment:
Instead of these `.version*` files living in the root directory, it would be nicer to have them in their own directory like `reports/`